File: misc/cut.lua
local w = weechat
local g = {
    script = {
        name = "cut",
        version = "0.1",
        author = "amosbird <amosbird@gmail.com>",
        license = "WTFPL",
        description = "Cut command buffer"
    }
}

function setup()
    assert(
        w.register(
            g.script.name,
            g.script.author,
            g.script.version,
            g.script.license,
            g.script.description,
            "", ""),
        "Unable to register script. Perhaps it's already loaded before?")

    local wee_ver = tonumber(w.info_get("version_number", "") or 0)
    if wee_ver < 0x01000000 then
        error("This script requires WeeChat v1.0 or higher")
    end
    w.hook_command(
        g.script.name,
        "Cut the buffer string into system's clipboard",
        "", "", "", "command_cb", "")
end

function cmd_cb(a, b, cmd, rc, out, err)
    w.print("", a)
    w.print("", b)
    w.print("", cmd)
    w.print("", rc)
    w.print("", out)
    w.print("", err)
end

function command_cb(_, buffer, _)
    line = w.buffer_get_string(buffer, "input")
    input_pos = w.buffer_get_integer(buffer, "input_pos")
    input_line = string.sub(line, 0, input_pos + 1)
    -- w.hook_process("bash -c 'echo " .. input_line .. " | osc52clip'", 10 * 1000, "", "")
    local f = assert(io.open("/tmp/s", "w"))
    assert(f:write(input_line))
    assert(f:close())
    assert(os.execute("osc52clip < /tmp/s"))
    w.buffer_set(buffer, "input", string.sub(line, input_pos + 1))
    w.buffer_set(buffer, "input_pos", "0")
    return w.WEECHAT_RC_OK
end

setup()

File: misc/cut.pl
use strict;
use utf8;
use MIME::Base64;
use Encode;

if (!weechat::register ("cut", "Amos Bird", "0.1", "WTFPL", "Cut command buffer", "", ""))
{
    weechat::print ("", "\tCut is already loaded");
    return weechat::WEECHAT_RC_OK;
}
else
{
    weechat::hook_command("Cut", "Cut the buffer string into system's clipboard", "", "", "", "command_cb", "");
}

sub command_cb {
    my ($data, $buffer, $args) = @_;
    my $line = decode("utf8", weechat::buffer_get_string($buffer, "input"));
    my $input_pos = weechat::buffer_get_integer($buffer, "input_pos");
    my $input_line = substr($line, 0, $input_pos);
    my $s = encode_base64(encode("utf8", $input_line), "");
    my $osc52 = "\033]52;c;".$s."\07";
    my $esc = "\033Ptmux;\033".$osc52."\033\\"; # tmux
    open(TTY, "> /dev/stdout");
    *STDOUT = *TTY;
    print $esc;
    close(TTY);
    weechat::buffer_set($buffer, "input", encode("utf8", substr($line, $input_pos)));
    weechat::buffer_set($buffer, "input_pos", "0");
    return weechat::WEECHAT_RC_OK
}
